tests/nix-shell.sh

```diff
@@ -58,6 +58,7 @@ output=$($TEST_ROOT/shell.shebang.rb abc ruby)
 
 # Test 'nix develop'.
 nix develop -f shell.nix shellDrv -c bash -c '[[ -n $stdenv ]]'
+echo foo | nix develop -f shell.nix shellDrv -c cat  # preserve stdin with `-c`
```

Comment:
```suggestion
echo foo | nix develop -f shell.nix shellDrv -c cat  # preserve stdin with `-c` | grep -q foo
```

(to make sure that the command is actually ran)

Also, this might also #4228 . If it does, do you mind also adding a test-case for it? Something like

```sh
# Ensure that `nix develop --command` actually executes the command if stdout isn't a terminal
echo "" | nix develop nixpkgs#hello --command echo foo |& grep -q foo
```

Reply:
Hah, indeed it does! Added.